src/controller/src/lib.rs

```diff
@@ -275,7 +276,7 @@ impl<T: Timestamp> Controller<T> {
 
 impl<T> Controller<T>
 where
-    T: TimestampManipulation,
+    T: TimestampManipulation + ComputeControllerTimestamp,
```

Comment:
Initially, I didn't remove this because I was thinking that the Storage Controller also needs it, in which case it might be good to still list it. But maybe it's enough to do this only at the big impl that is at the bottom of this file.